conjureup/models/provider.py

```diff
@@ -445,19 +460,37 @@ def __init__(self):
             widget=PasswordEditor(),
             key='password'
         )
-        self.external_network = Field(
-            label='external network',
-            widget=StringEditor(),
-            key='external-network',
-            storable=False
-        )
+
+    def login(self, credential):
+        if app.vsphere.authenticated:
+            return
+
+        app.vsphere.client = VSphereClient(**credential)
+
+        try:
+            app.vsphere.client.login()
+            app.vsphere.authenticated = True
+        except:
+            raise Exception(
+                "Could not log in to VSphere, please "
+                "check your credentials and try again.")
```

Comment:
Another bare `except` and this also completely masks the original error.  We should at a minimum log what caused the failure, but we probably also want to distinguish between actual auth errors, connection errors, and other types of errors.